tests/unit/Framework/Attributes/RetryTest.php

```diff
@@ -0,0 +1,108 @@
+<?php declare(strict_types=1);
+/*
+ * This file is part of PHPUnit.
+ *
+ * (c) Sebastian Bergmann <sebastian@phpunit.de>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+namespace PHPUnit\Framework\Attributes;
+
+use DateTime;
+use Exception;
+use LogicException;
+use PHPUnit\Framework\TestCase;
+
+final class RetryTest extends TestCase
```

Comment:
Currently, it seems pre/post test hooks are not called. But that can be an issue.

I would like to have this tested by E2E tests to show the behaviour and also with other things like process isolation.